### PR TITLE
refactor: integrate ws hooks into api hooks

### PR DESCRIPTION
**Pull request by dialexo**
This PR improves on #48 by integrating the websocket hooks into the API hooks directly, each with a boolean allowing consumer to enable websocket updates

Front-end applications that still use the previously `ws` exported member should update (Compose, Perform, ...).
Example:

```ts
// before
const item = useItem(itemId)
ws.hooks.useItemUpdates()
```

```ts
// should now be
const item = useItem(itemId)

// specific components can also optionally opt-out from receiving websocket updates:
const item = useItem(itemId, {getUpdates: false})
```

### CODE REVIEW

File: src/hooks/item.ts [hunk 1]
```diff
@@ -97,8 +123,15 @@ export default (queryClient: QueryClient, queryConfig: QueryClientConfig) => {
...
 
-    useSharedItems: () =>
-      useQuery({
+    useSharedItems: (options?: { getUpdates?: boolean }) => {
+      const getUpdates = options?.getUpdates ?? enableWebsocket;
+
+      const { data: currentMember } = useCurrentMember();
+      itemWsHooks?.useSharedItemsUpdates(
+        getUpdates ? currentMember?.get('id') : null,
+      );
```
Comment (pyphilia) [inline src/hooks/item.ts:132]: Btw is there an issue here if the `currentMember` takes time to load and is `undefined` first for some time?

**Review by pyphilia — APPROVED** [review]
Great!!

File: src/hooks/item.ts (hunk 1 above)
Comment (dialexo) [reply src/hooks/item.ts:132]: I've wondered the same but I don't think it is an issue. `useCurrentMember` may return undefined as data first (and `isLoading` set to true) but the rest of the hook will still execute (in this case the shared items will still get fetched from the query client) and the WS hook will be called with null and do nothing. I believe when `useCurrentMember` finally returns the actual `currentMember`, the hook will be recalled and that time `getUpdates ? currentMember?.get('id')` will return the actual id and perform the WS subscription

Comment (pyphilia) [reply src/hooks/item.ts:132]: I think so as well, but I wanted to confirm with you^^

**Pull request merged**